boskos/ranch/ranch.go

```diff
@@ -142,29 +143,35 @@ func (r *Ranch) Acquire(rType, state, dest, owner, requestID string) (*common.Re
 	typeCount := 0
 	for idx := range resources {
 		res := resources[idx]
-		if rType == res.Type {
-			typeCount++
-			if state == res.State && res.Owner == "" {
-				matchingResoucesCount++
-				if matchingResoucesCount >= rank {
-					logger = logger.WithField("resource", res.Name)
-					res.Owner = owner
-					res.State = dest
-					logger.Debug("Updating resource.")
-					updatedRes, err := r.Storage.UpdateResource(res)
-					if err != nil {
-						logger.WithError(err).Errorf("could not update resource %s", res.Name)
-						return nil, err
-					}
-					// Deleting this request since it has been fulfilled
-					if requestID != "" {
-						logger.Debug("Cleaning up requests.")
-						r.requestMgr.Delete(ts, requestID)
-					}
-					logger.Debug("Successfully acquired resource.")
-					return &updatedRes, nil
-				}
+		if rType != res.Type {
+			continue
+		}
+		typeCount++
+
+		if state != res.State {
+			continue
+		}
+		if res.Owner != "" {
+			continue
+		}
+		matchingResoucesCount++
+		if matchingResoucesCount >= rank {
```

Comment:
Why no `if matchingResourcesCount < rank { continue }` to keep the flow consistent?